plugins/woocommerce/tests/php/includes/abstracts/class-wc-abstract-product-test.php

```diff
@@ -51,12 +65,95 @@ public function test_fetching_of_approved_downloads() {
 			'If a trusted download directory rule is disabled, corresponding product downloads will also be marked as disabled.'
 		);
 
-		$download_directories->set_mode( Download_Directories::MODE_DISABLED );
+		$this->download_directories->set_mode( Download_Directories::MODE_DISABLED );
 
 		$this->assertCount(
 			2,
-			wc_get_product( $product->get_id() )->get_downloads(),
+			wc_get_product( $this->product->get_id() )->get_downloads(),
 			'Disabling the Approved Download Directories system entirely does not impact our ability to fetch product downloads.'
 		);
 	}
+
+	/**
+	 * @testdox Confirm that when product downloads are set, the operation is successful (or else errors are raised) as appropriate.
+	 */
+	public function test_setting_of_product_downloads() {
```

Comment:
Not a biggie, just a question---I'd write these tests into separate methods, since they all test different aspects of the same feature and the purist in me says that each test method should test only one thing. What do you think?

Reply:
It's a little funny because I originally wrote it that way, then ditched the `expectException()` approach at the end of each individual test method and consolidated them here because  I felt it told the story of what we are trying to do a little more fluently.

*But,* on re-reflection ... you're probably right. Will update :+1:

Reply:
Updated (@peterfabian).

Reply:
😅 Apologies for the back and forth, then!